.github/workflows/build.yml

```diff
@@ -0,0 +1,57 @@
+name: build
+on:
+  push:
+    branches:
+    - master
+    - v*
+    tags:
+    - v*
+  pull_request:
+jobs:
+  build_push_image:
+    name: Build and push image
+    runs-on: ubuntu-latest
+    if: ${{ startsWith(github.ref, 'refs/heads/') || startsWith(github.ref, 'refs/tags/') }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    # For multi-platform support
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Declare branch
+      run: |
+        echo "sha_short=$(git rev-parse --short "$GITHUB_SHA")" >> "$GITHUB_ENV"
+        echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_ENV"
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
```

Comment:
Just confirm, you will add secrets `DOCKER_USERNAME` and `DOCKER_PASSWORD` somewhere in github action, right?

Reply:
Yes, already setup in organization level secrets.

Reply:
thanks for confirmation.